File: python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self.rows = [list(map(int, row.split(' '))) for row in matrix_string.split('\n')]
        self.columns = [[row[i] for row in self.rows] for i in range(len(self.rows[0]))]

    def row(self, index):
        return self.rows[index]

    def column(self, index):
        return self.columns[index]

File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrix) {
    this.rows = matrix.split('\n')
      .map(row => row.split(' ')
      .map(r => Number(r)));
    this.columns = this.rows.reduce((columns, row, i) => {
      row.forEach((element, j) => {
        if (i == 0) {
          columns.push([row[j]]);
        } else {
          columns[j].push(element);
        }
      });
      return columns;
    }, []);
  }
}